lib/services/arweave/arweave_service.dart

```diff
@@ -836,6 +837,27 @@ class ArweaveService {
 
     return response.data?['arweave']['usd'];
   }
+
+  // instantiate entity from tx and metadata
+  Future<Uint8List> entityMetadataFromFromTxId(
+    String txId,
+    SecretKey? driveKey,
+  ) async {
+    // final http = ArDriveHTTP();
+    // final url = '${client.api.gatewayUrl.origin}/$txId';
+    // print('[ArweaveService] Getting metadata for $txId - (RUL: $url))');
+    // final response = await http.getAsBytes(url);
+    // final Uint8List metadata = response.data;
+    // return utf8.decode(metadata);
+
+    // The code below will eat all the redirects. How do I get the final URL?
```

Comment:
TODO! - can we use the methods on the arweave library to generate the URL? do we have a method that already uses it somewhere in the code?
```suggestion
    // The code above will eat all the redirects. How do I get the final URL?
```

Reply:
77ad10ad65e9e83b8a0cc6f2a2628442c2686d48